simulator/src/rtmc/mod.rs

```diff
@@ -52,8 +53,11 @@ pub fn run() -> Result<(), Error> {
             blobs.push(std::fs::read(path).map_err(Error::IoError)?);
         }
     }
-    let transaction = compile(&content, &network, BlobProvider::new_with_blobs(blobs))
-        .map_err(Error::CompileError)?;
+    let transaction =
+        compile(&content, &network, BlobProvider::new_with_blobs(blobs)).map_err(|err| {
+            eprintln!("{}", compile_error_diagnostics(&content, err));
```

Comment:
This part looks a bit hacky. But didn't have any other idea how to implement it in a nicer way.